edsdme/blocks/search/SearchCards.js

```diff
@@ -49,31 +48,79 @@ export default class Search extends PartnerCards {
       </div>`;
   }
 
-  setContentTypeCounts() {
-    const counts = { countAll: 0, countAssets: 0, countPages: 0 };
-    const filteredCards = [];
+  // eslint-disable-next-line consistent-return
+  async getCards() {
+    const url = new URL(
+      'https://14257-dxpartners-stage.adobeioruntime.net/api/v1/web/dx-partners-runtime/search-apc/search-apc?',
+    );
 
-    this.cards.forEach((card) => {
-      if (card.contentArea?.contentType === 'asset' || card.contentArea?.contentType === 'page') {
-        counts.countAll += 1;
-        if (card.contentArea.contentType === 'asset') counts.countAssets += 1;
-        if (card.contentArea.contentType === 'page') counts.countPages += 1;
-        filteredCards.push(card);
-      }
-    });
+    const startCardIndex = (this.paginationCounter - 1) * this.cardsPerPage;
 
-    this.cards = filteredCards;
-    this.contentTypeCounter = counts;
-  }
+    const partnerDataCookie = getPartnerDataCookieObject(getCurrentProgramType());
+    const partnerLevel = partnerDataCookie?.level || 'public';
+    const regions = partnerDataCookie?.level || 'worldwide';
+
+    const { locales } = getConfig();
+    const localesData = getLocale(locales);
+
+    const queryParams = new URLSearchParams(url.search);
+    queryParams.append('partnerLevel', partnerLevel);
+    queryParams.append('regions', regions);
+    queryParams.append('type', this.contentType);
+    queryParams.append('term', this.searchTerm);
+    queryParams.append('geo', localesData.prefix && localesData.region);
+    queryParams.append('language', localesData.ietf);
+    queryParams.append('from', startCardIndex.toString());
+    queryParams.append('size', this.cardsPerPage);
+    const sortMap = { 'most-recent': 'recent', 'most-relevant': 'relevant' };
+    queryParams.append('sort', sortMap[this.selectedSortOrder.key]);
+
+    const postData = {
+      filters: {
```

Comment:
@SonjaPopovic shouldn't we assign filters in a way that doesn't care about the keys names, since they are authorable (`type`, `product`, `topic`).

The only exception is `language` which must be called that due to a mapping in the backend.

Reply:
Ok, thanks, was not avare of that requirement, i thought we have harcoded filters. 
Than i will make this flexible for filters.
For language, i guess since it can also be not selected by user, you will handle it on backand to allways include it in that mapping ? It sounds little confusing to use this as mapping since it can be empty, maybe language query param is used for that mapping, and not this fitler from req body?